openmrs/src/main/java/org/motechproject/openmrs/resource/impl/BaseResource.java

```diff
@@ -85,6 +88,9 @@ protected void delete(Config config, String path, Object... params) {
 
     private ResponseEntity<String> exchange(Config config, URI url, HttpMethod method, String body, HttpHeaders headers) {
         headers.add("Authorization", "Basic " + prepareCredentials(config));
+
+        httpClient.getState().clearCookies();
```

Comment:
Ok thx for research. I think it will be better to avoid changing versions or adding new dependency just before 0.29 release. Especially that httpClient/RestTemplate is really often used in OpenMRS module. So far this solution looks good to me. However we should consider sth better in the future